rootfs/etc/s6/mariadb/run

```diff
@@ -29,18 +29,17 @@ then
   echo "DROP DATABASE IF EXISTS test;" >> ${INIT_FILE}
   echo "FLUSH PRIVILEGES;" >> ${INIT_FILE}
 
-  if [[ -n "${MARIADB_USERNAME}" && -n "${MARIADB_DATABASE}" ]]
+  MARIADB_DATABASE=${MARIADB_DATABASE:-$MARIADB_USERNAME}
+  if [ ! -z "${MARIADB_DATABASE}" ]
   then
-    echo "CREATE DATABASE IF NOT EXISTS \`${MARIADB_USERNAME}\`;" >> ${INIT_FILE}
-    echo "GRANT ALL PRIVILEGES ON \`${MARIADB_DATABASE}\`.* TO '${MARIADB_USERNAME}'@'%' IDENTIFIED BY '${MARIADB_PASSWORD}';" >> ${INIT_FILE}
-    echo "FLUSH PRIVILEGES;" >> ${INIT_FILE}
-  fi
-
-  if [[ -n "${MARIADB_USERNAME}" && -z "${MARIADB_DATABASE}" ]]
-  then
-    echo "CREATE DATABASE IF NOT EXISTS \`${MARIADB_USERNAME}\`;" >> ${INIT_FILE}
-    echo "GRANT ALL PRIVILEGES ON \`${MARIADB_USERNAME}\`.* TO '${MARIADB_USERNAME}'@'%' IDENTIFIED BY '${MARIADB_PASSWORD}';" >> ${INIT_FILE}
-    echo "FLUSH PRIVILEGES;" >> ${INIT_FILE}
+    echo "CREATE DATABASE IF NOT EXISTS \`${MARIADB_DATABASE}\`;" >> ${INIT_FILE}
+
+    if [ ! -z "${MARIADB_USERNAME}" ]
+    then
+      echo "CREATE USER IF NOT EXISTS '${MARIADB_USERNAME}'@'%' IDENTIFIED BY '${MARIADB_PASSWORD}';" >> ${INIT_FILE} # is able to create an User with NO Password
```

Comment:
Just understand I'm a picky person and this is nothing I'm earning my money with. So it can take some time. If you don't like it's fine to me.